### DISCUSSION TITLE
Help with flashing the mailbox sensor using UNIProg

### DISCUSSION THREAD

**Discussion by RolfSzimnau** [PricelessToolkit/MailBoxGuard#42] [General]
Utilizing the UNIProg Programmer, I intend to upload the Mailbox_Guard_Sensor.ino firmware onto the connected board via USB. My development environment is Arduino IDE 2.2.1, with the addition of http://drazzy.com/package_drazzy.com_index.json in the preferences. However, I'm encountering difficulties locating megaTinyCore from Spence Konde in the library manager, as per the guidance at https://wolles-elektronikkiste.de/en/using-megatinycore. I've diligently followed the outlined steps but seem to be missing something. I've ensured the upload of all necessary files to \Arduino\Arduino_libraries.

I could use some assistance in troubleshooting and getting on the right track. Can anyone provide guidance and let me know if I'm overlooking any crucial details? Please inform me if you need additional information to assist me effectively.

**Comment by Credo102** [comment]
I could not get it to work with IDE 2+ either then I noticed.  Arduino Setup "IDE 2.0.x unsupported" on the read me.
I got it to work with an earlier Arduino 1.8??
And made sure my setup looked like the screenshot.

Hope that helps

**Reply by PricelessToolkit** [reply]
Yes IDE 2+ is not supported.
Install IDE v1.8 and add this link "http://drazzy.com/package_drazzy.com_index.json" to the additional Board manager and install megaTinyCore package from the Board manager, select board settings using a screenshot and set programmer type Serial2UPDI. Connect UNIProg with 3 cables 3.3v, GND, UPDI then click upload.

Thats all.

**Comment by PricelessToolkit** [comment]
It is very strange !, something's wrong with your Arduino installation. Reinstalled megaTinyCore, and everything works for me. I don't even have an idea what is going on. PS: From the Screenshot, I can see your Arduino installation is on OneDrive ? or is it just naming?

**Comment by RolfSzimnau** [comment]
I got it working; thank you both for the assistance. Under my profile, I had an Arduino folder that persisted even after uninstalling. So, this morning, I removed everything and performed a fresh install. After the installation, I encountered no issues—everything worked on the first attempt. Thank you both. One last question: I am using WhatsApp, but is it also possible to combine this with the OLED? This way, if movement is detected, the screen would be activated as well?

**Reply by PricelessToolkit** [reply]
For adding Text on Oled first test if you receive a WhatsApp message.

then change this part in the Gateway code 

```c

        if (recv == NewMailCode){
            sendMessage("New Mail in The Mailbox");
         }
        
         if (recv == LowBatteryCode){
            sendMessage("MailBox Battery is LOW!");
         }

```

To

```c

        if (recv == NewMailCode){
            sendMessage("New Mail in The Mailbox");
            display.setFont(ArialMT_Plain_16);
            display.setTextAlignment(TEXT_ALIGN_CENTER);
            display.drawString(64, 15, "NEW MAIL");
            display.display();
         }
        
         if (recv == LowBatteryCode){
            sendMessage("MailBox Battery is LOW!");
            display.setFont(ArialMT_Plain_16);
            display.setTextAlignment(TEXT_ALIGN_CENTER);
            display.drawString(64, 30, "LOW BATT");
            display.display();
         }

```
I have not tested it but it will work, maybe the text position will need adjustment `display.drawString(75, 25, "LOW BATT");` The first number is horizontal position second is vertical.

To clear the message from the screen just push the reset button.

**Reply by RolfSzimnau** [reply]
I've managed to get the above working, thank you. My question is: Is there any hope that, at some point, you will create a merge between WhatsApp and OLED INO? Currently, it only shows the text on the LilyGO. However, if possible, it would be nice to have the graphics also displayed. I've attempted to merge the two, but I'm not a proficient coder. Actually i suck at it...

**Reply by PricelessToolkit** [reply]
I suck at it too :) To be honest, you have a better chance to get into a smart home and use MQTT version, than I will make a new firmware. I don't have much time.

**Reply by RolfSzimnau** [reply]
Thanks for replying; I totally understand. I will play around a little. And if, by miracle, I get it working, I will let you know. Have a great evening.
